packages/opentelemetry-resources/src/ResourceImpl.ts

```diff
@@ -120,28 +124,35 @@ class ResourceImpl implements Resource {
     return this._rawAttributes;
   }
 
+  public getSchemaUrl(): string | undefined {
+    return this._schemaUrl;
+  }
+
   public merge(resource: Resource | null): Resource {
     if (resource == null) return this;
 
     // Order is important
     // Spec states incoming attributes override existing attributes
-    return ResourceImpl.FromAttributeList([
-      ...resource.getRawAttributes(),
-      ...this.getRawAttributes(),
-    ]);
+    const mergedSchemaUrl = mergeSchemaUrl(this, resource);
+    return ResourceImpl.FromAttributeList(
+      [...resource.getRawAttributes(), ...this.getRawAttributes()],
+      mergedSchemaUrl
+    );
   }
 }
 
 export function resourceFromAttributes(
-  attributes: DetectedResourceAttributes
+  attributes: DetectedResourceAttributes,
+  schemaUrl?: string
```

Comment:
instead of putting a positional argument here, please put an options type that we can extend it later, so a user would use it like this:

`resourceFromAttributes({ 'foo': 'foo-value' }, { schemaUrl: 'https://example.com/schema/v1.0.0' });`

This way, if we ever have to extend the amount of options we can do that more easily. 🙂

Reply:
updated :)